justfile

```diff
@@ -0,0 +1,242 @@
+            echo "SpacetimeDB failed to start. See {{stdb_log}}"; \
+            exit 1; \
+        fi; \
+    fi
+
+# Stop SpacetimeDB
+spacetime-stop:
+    @if [ -f "{{stdb_pid}}" ] && kill -0 "$(cat "{{stdb_pid}}")" 2>/dev/null; then \
+        kill "$(cat "{{stdb_pid}}")"; \
+        rm -f "{{stdb_pid}}"; \
+        echo "SpacetimeDB stopped"; \
+    else \
+        rm -f "{{stdb_pid}}"; \
+        echo "SpacetimeDB not running"; \
+    fi
+
+# Publish the strategic module (optional target can be provided: just publish target=localhost)
+publish server=spacetime_url:
+    @cd "{{strategic_dir}}" && spacetime publish --server {{server}} {{spacetime_module}}
+
+# Publish and clear the module database
+publish-reset server=spacetime_url:
+    @cd "{{strategic_dir}}" && spacetime publish --delete-data=always --server {{server}} {{spacetime_module}}
+
+# Serve the strategic UI locally
+serve-ui:
+    @mkdir -p "{{run_dir}}"
+    @if [ -f "{{http_pid}}" ] && kill -0 "$(cat "{{http_pid}}")" 2>/dev/null; then \
+        echo "UI server already running (pid $(cat "{{http_pid}}"))"; \
+    else \
+        rm -f "{{http_pid}}"; \
+        python3 -m http.server {{ui_port}} --bind 127.0.0.1 --directory "{{strategic_static}}" >"{{http_log}}" 2>&1 & \
+        echo $! > "{{http_pid}}"; \
+        sleep 1; \
+        echo "UI server running on http://localhost:{{ui_port}}/map.html"; \
+    fi
+
+# Serve the strategic UI on all interfaces (for VPS/public use)
+serve-ui-public:
+    @mkdir -p "{{run_dir}}"
+    @if [ -f "{{http_pid}}" ] && kill -0 "$(cat "{{http_pid}}")" 2>/dev/null; then \
+        echo "UI server already running (pid $(cat "{{http_pid}}"))"; \
+    else \
+        rm -f "{{http_pid}}"; \
+        python3 -m http.server {{ui_port}} --bind {{public_bind}} --directory "{{strategic_static}}" >"{{http_log}}" 2>&1 & \
+        echo $! > "{{http_pid}}"; \
+        sleep 1; \
+        echo "UI server running on http://{{public_bind}}:{{ui_port}}/map.html"; \
+    fi
+
+# Stop the strategic UI server
+stop-ui:
+    @if [ -f "{{http_pid}}" ] && kill -0 "$(cat "{{http_pid}}")" 2>/dev/null; then \
+        kill "$(cat "{{http_pid}}")"; \
+        rm -f "{{http_pid}}"; \
+        echo "UI server stopped"; \
+    else \
+        rm -f "{{http_pid}}"; \
+        echo "UI server not running"; \
+    fi
+
+# Open the strategic UI in a browser
+open-ui:
+    @url="http://localhost:{{ui_port}}/map.html"; \
+    if command -v xdg-open >/dev/null 2>&1; then \
+        xdg-open "$$url" >/dev/null 2>&1; \
+    elif command -v open >/dev/null 2>&1; then \
+        open "$$url" >/dev/null 2>&1; \
+    elif command -v firefox >/dev/null 2>&1; then \
+        firefox "$$url" >/dev/null 2>&1; \
+    elif command -v google-chrome >/dev/null 2>&1; then \
+        google-chrome "$$url" >/dev/null 2>&1; \
+    else \
+        echo "Open $$url"; \
+    fi
```

Comment:
Could also benefit from [`which`](https://just.systems/man/en/functions.html#executables). And all of the other optional executables down the script as well